src/gui/gui_province_window.hpp

```diff
@@ -1881,6 +1881,13 @@ class province_view_window : public window_element_base {
 		}
 	}
 
+	void on_update(sys::state& state) noexcept override {
+		/* Properly updates the window when ownership changes - prevents weird bugs */
+		if(state.province_ownership_changed.load(std::memory_order_acquire) == true) {
```

Comment:
this is not a great fix, because the main loop is going to clear that ASAP, meaning that this will only work if the ui update happens to run in a quite narrow window